### PR TITLE
Add cra

### PR DESCRIPTION
**Pull request by dbertella**
So this is basically a complete rewrite. I started from CRA and installed storybook on top. 
Everything seems to work fine, tests included so I think we could switch to this. 

I added my version of the toggle but feel free to update it as you wish

### CODE REVIEW

File: src/components/Toggle.tsx [hunk 1]
```diff
@@ -1,139 +1,121 @@
...
-          }
-        : {}),
-    }}
-    {...rest}
-  >
-    <span
+export const Body: FC<
+  CheckDisable & { onChange: (e: ChangeEvent<HTMLInputElement>) => void }
+> = ({ checked, disabled, children, onChange }) => (
+  <Fragment>
```
Comment (dbertella) [inline src/components/Toggle.tsx:39]: Why using Fragment instead of `<>`?

File: src/components/Toggle.tsx [hunk 2]
```diff
@@ -1,139 +1,121 @@
...
+        borderRadius: '0.875rem',
+        transitionDelay: '0.12s',
+        transitionDuration: '0.2s',
+        transitionProperty: 'background, border',
+        transitionTimingFunction: 'cubic-bezier(0, 0, 0.2, 1)',
+        position: 'relative',
+        border: '1px solid transparent',
+        backgroundColor: 'lightGray',
+        padding: 0,
+        ...ifStyle(disabled, { borderColor: 'darkGray' }),
```
Comment (dbertella) [inline src/components/Toggle.tsx:69]: this one I will call it `ifProp` anyway I still don't like the spread

File: src/components/Toggle.tsx [hunk 3]
```diff
@@ -1,139 +1,121 @@
...
   function handleChange(event: ChangeEvent<HTMLInputElement>): void {
-    onChange?.(event.currentTarget.checked);
+    onChange(event.currentTarget.checked)
   }
 
+  const cd: CheckDisable = { checked, disabled }
+
   return (
-    <Label disabled={disabled}>
+    <LabelContainer {...cd}>
```
Comment (dbertella) [inline src/components/Toggle.tsx:114]: here you need to spread rest as well and also in the component

File: src/components/Toggle.tsx [hunk 4]
```diff
@@ -1,139 +1,121 @@
...
-        checked={checked}
-        onChange={handleChange}
-      />
-
-      <ToggleHandler disabled={disabled} checked={checked} />
-    </Label>
-  );
-};
+      <Body {...cd} onChange={handleChange}>
+        <Handler {...cd} />
```
Comment (dbertella) [inline src/components/Toggle.tsx:117]: why keeping Handler in a separate component than Body? Also Body it has a strange name unless we keep this name as a convention not sure

File: src/components/Toggle.tsx (hunk 1 above)
Comment (nkint) [reply src/components/Toggle.tsx:39]: With `<>` I get the error `JSX fragment is not supported when using an inline JSX factory pragmats(17017)`.

Do you have it?
Not really sure what does it means... any ideas?

File: src/components/Toggle.tsx (hunk 2 above)
Comment (nkint) [reply src/components/Toggle.tsx:69]: I don't like spreading too.. let's think about it...

File: src/components/Toggle.tsx (hunk 3 above)
Comment (nkint) [reply src/components/Toggle.tsx:114]: 👍

File: src/components/Toggle.tsx (hunk 1 above)
Comment (dbertella) [reply src/components/Toggle.tsx:31]: is going to be some config

**Pull request merged**